pynblint/nb_linting.py

```diff
@@ -429,3 +432,59 @@ def get_bottom_md_lines_ratio(notebook, bottom_size=4):
         return 0
     else:
         return md_bottom_cells / (md_first_cells + md_bottom_cells)
+
+
+def is_untitled(notebook):
+    """
+    The function takes a notebook and checks whether it has a title or not
+
+        Args:
+            notebook(Notebook): python object representing the notebook
+        Returns:
+            boolean: True if the name of the notebook is Untitled, False otherwise
+
+        A way you might use me is
+
+        untitled = is_untitled(notebook)
+    """
+    if os.path.basename(notebook.path) == "Untitled.ipynb":
+        return True
+    else:
+        return False
+
+
+def is_filename_charset_restricted(notebook):
+    """
+    The function takes a notebook and checks whether it has a title in the [A-Za-z0-9_.-] charset
+
+        Args:
+            notebook(Notebook): python object representing the notebook
+        Returns:
+            boolean: True if the name of the notebook is withing the charset, False otherwise
+
+        A way you might use me is
+
+        restricted_filename = is_filename_charset_restricted(notebook)
+    """
+    if re.search("^[A-Za-z0-9_.-]+$", os.path.basename(notebook.path)):
+        return True
+    else:
+        return False
+
+def is_filename_short(notebook, filename_max_length = 25):
```

Comment:
It just looked like a reasonable number. Not very scientific by me. Pymentel is not that much deterministic on this point: he just says 
> 1. Use short titles with a restrict charset (A-Z a-z 0-9 .
-) for notebook files and markdown headings for more
detailed ones in the body. As discussed in Section IV-B,
some operating systems may not support characters that many
notebook titles use. Since notebooks support markdown, we
recommend using it to write the complex titles inside the
notebooks and leave the notebook title as simple as possible.

Reply:
Defining best practices based on numbers in a scientific way is not easy. Many linters end up using opinionated "magic numbers" (e.g., 80 characters as the maximum length of a code line), but trying to suggest a rationale for them (e.g., the typical length of a terminal line is 80 characters) or other pragmatic reasons; either way, these numbers are always configurable, and this is what matters the most, I think.